File: applications/frontend/src/types/gitops.ts
export interface GitOpsApp {
  id: number;
  created_at: string;
  updated_at: string;
  name: string;
  repo_url: string;
  branch: string;
  path: string;
  namespace: string;
  sync_status: 'Synced' | 'OutOfSync' | 'Unknown';
  last_synced: string | null;
}

export interface CreateGitOpsAppRequest {
  name: string;
  repo_url: string;
  branch: string;
  path: string;
  namespace: string;
}

File: applications/backend/internal/models/gitops.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type GitOpsApp struct {
	ID          uint           `gorm:"primarykey" json:"id"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
	Name        string         `gorm:"not null" json:"name"`
	RepoURL     string         `gorm:"not null" json:"repo_url"`
	Branch      string         `gorm:"default:main" json:"branch"`
	Path        string         `json:"path"` // Path to manifests in repo
	Namespace   string         `json:"namespace"`
	SyncStatus  string         `gorm:"default:Unknown" json:"sync_status"` // Synced, OutOfSync
	LastSynced  *time.Time     `json:"last_synced"`
}
